region/norfair/east/Upper Norfair Farming Room.json

```diff
@@ -139,6 +143,146 @@
         "h_canHeatedCrystalFlash"
       ]
     },
+    {
+      "link": [1, 2],
+      "name": "Come in Shinecharging, Leave Shinecharged",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 12,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "Wave",
+        "canShinechargeMovementTricky",
+        {"heatFrames": 200}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 20
+        }
+      },
+      "unlocksDoors": [{"types": ["ammo"], "requires": ["never"]}]
+    },
+    {
+      "link": [1, 2],
+      "name": "Leave Shinecharged (In-Room Runway)",
+      "requires": [
+        {"or": [
+          "Wave",
+          {"obstaclesCleared": ["A"]}
+        ]},
+        {"or": [
+          {"canShineCharge": {
+            "usedTiles": 12,
+            "openEnd": 1
+          }},
+          {"and": [
+            {"canShineCharge": {
+              "usedTiles": 13,
+              "openEnd": 1
+            }},
+            {"doorUnlockedAtNode": 1}  
+          ]}
+        ]},
+        "canShinechargeMovementTricky",
+        {"heatFrames": 250}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 20
+        }
+      },
+      "unlocksDoors": [{"types": ["ammo"], "requires": [{"obstaclesCleared": ["A"]}]}]
```

Comment:
Should a missing line for the unlockDoors at 1 cause the test to fail?

Reply:
It should be fine because the strat starts at 1, so it's assumed an implicit door unlock strat could be done before doing the strat.